llama_stack/apis/datasets/datasets.py

```diff
@@ -50,13 +66,29 @@ class Datasets(Protocol):
     @webmethod(route="/datasets", method="POST")
     async def register_dataset(
         self,
-        dataset_id: str,
-        dataset_schema: Dict[str, ParamType],
-        url: URL,
-        provider_dataset_id: Optional[str] = None,
-        provider_id: Optional[str] = None,
+        schema: Schema,
+        uri: str,
+        uri_params: Optional[Dict[str, Any]] = None,
```

Comment:
why separate it out? why couldn't the user embed these params in the `uri` itself?

Reply:
Yeah, I think having embed these params in `uri` also make sense. E.g. `huggingface://tatsu-lab/alpaca?split=train`

cc @hardikjshah  ok we remove uri_params?

Reply:
removed `uri_params` and introduced `DataSource`.